src/Application/ApplicationFileProcessor.php

```diff
@@ -52,12 +53,17 @@ public function __construct(
         private readonly FileProcessor $fileProcessor,
         private readonly ArrayParametersMerger $arrayParametersMerger,
         private readonly MissConfigurationReporter $missConfigurationReporter,
+        private readonly FileDependenciesCache $fileDependenciesCache,
     ) {
     }
 
     public function run(Configuration $configuration, InputInterface $input): ProcessResult
     {
-        $filePaths = $this->filesFinder->findFilesInPaths($configuration->getPaths(), $configuration);
+        $allFiles = $this->filesFinder->findFilesInPaths($configuration->getPaths(), $configuration);
```

Comment:
We split this function in two, first we find all possible files to process and then we filter them using the cache